src/EGraphs/saturation.jl

```diff
@@ -90,24 +89,23 @@ function eqsat_search!(
       ids_left = cached_ids(g, rule.left)
       for i in ids_left
         cansearch(scheduler, rule_idx, i) || continue
-        n_matches += rule.ematcher_left!(g, rule_idx, i, rule.stack, ematch_buffer)
-        inform!(scheduler, rule_idx, i, n_matches)
+        eclass_matches = rule.ematcher_left!(g, rule_idx, i, rule.stack, ematch_buffer)
+        n_matches += eclass_matches
+        inform!(scheduler, rule_idx, i, eclass_matches)
       end
 
       if is_bidirectional(rule)
         ids_right = cached_ids(g, rule.right)
         for i in ids_right
           cansearch(scheduler, rule_idx, i) || continue
-          n_matches += rule.ematcher_right!(g, rule_idx, i, rule.stack, ematch_buffer)
-          inform!(scheduler, rule_idx, i, n_matches)
+          eclass_matches = rule.ematcher_right!(g, rule_idx, i, rule.stack, ematch_buffer)
+          n_matches += eclass_matches
+          inform!(scheduler, rule_idx, i, eclass_matches)
         end
       end
```

Comment:
For bidirectional rules, the LHS and RHS may both match the same eclass. In this case inform! would be called twice for the same rule index and the same eclass in the same iteration. None of the implemented schedulers uses this, but it could be unexpected for people implementing their own scheduler.